src/Tests/Component/Signing/AuthnRequestSigningTest.php

```diff
@@ -0,0 +1,216 @@
+        $signatureIsVerified = $signatureVerifier->isSignedWith($authnRequestWithModifiedDataToSign, $certificate);
+
+        $this->assertFalse(
+            $signatureIsVerified,
+            'A signature that does not correspond with the data to sign and '
+            . 'how it is included in the http query should not be verified, but it is'
+        );
+    }
+
+    /**
+     * @test
+     * @group Signing
+     */
+    public function a_signature_is_not_verified_if_parameter_order_of_the_corresponding_http_query_is_not_correct()
+    {
+        $authnRequestWithModifiedDataToSign = $this->createSignedAuthnRequestPreparingDataToSignWith(
+            [$this, 'changeParameterAndEncodeDataToSign']
+        );
+
+        $certificate = SAML2_Certificate_X509::createFromCertificateData($this->getPublicKey());
+
+        $signatureVerifier   = new SignatureVerifier(new SAML2_Certificate_KeyLoader, new NullLogger);
+        $signatureIsVerified = $signatureVerifier->isSignedWith($authnRequestWithModifiedDataToSign, $certificate);
+
+        $this->assertFalse(
+            $signatureIsVerified,
+            'A signature that corresponds with the data to sign '
+            . 'but the data to sign is in the wrong order should not be verified, but it is'
+        );
+    }
+
+    /**
+     * @param array $params
+     * @return string
+     */
+    private function encodeDataToSignWithPhpsHttpBuildQuery(array $params)
+    {
+        return http_build_query($params);
+    }
+
+    /**
+     * @param array $params
+     * @return string
+     */
+    private function encodeDataToSignWithCustomEncoder(array $params)
+    {
+        $encodedParams = http_build_query($params);
+
+        return str_replace('%3A', ':', $encodedParams);
+    }
+
+    /**
+     * @param array $params
+     * @return string
+     */
+    private function changeParameterAndEncodeDataToSign(array $params)
+    {
+        return http_build_query(array_reverse($params, true));
+    }
+
+    /**
+     * @param callable $prepareDataToSign Expects an associative array of data to sign and returns a string to sign
+     * @param null|string $customSignature Signature to be used instead of signature to sign data to sign with
+     * @return AuthnRequest
+     */
+    private function createSignedAuthnRequestPreparingDataToSignWith(
+        callable $prepareDataToSign,
+        $customSignature = null
+    ) {
+        $keyData    = file_get_contents(__DIR__.'/../../../Resources/keys/development_privatekey.pem');
+        $privateKey = new XMLSecurityKey(XMLSecurityKey::RSA_SHA256, ['type' => 'private']);
+        $privateKey->loadKey($keyData);
+
+        $domDocument          = SAML2_DOMDocumentFactory::fromString($this->authRequestNoSubject);
+        $unsignedAuthnRequest = SAML2_AuthnRequest::fromXML($domDocument->firstChild);
+        $unsignedAuthnRequest->setSignatureKey($privateKey);
+        $unsignedAuthnRequest->setCertificates([$this->getPublicKey()]);
```

Comment:
I've done it this way, because I've seen it in SAML2 tests as well.
For instance, in the [MessageTest](https://github.com/simplesamlphp/saml2/blob/67ad5af84dc3215c437ccb396a7426f9f2de35b0/tests/SAML2/MessageTest.php) and the [SignatureValidationTest](https://github.com/simplesamlphp/saml2/blob/67ad5af84dc3215c437ccb396a7426f9f2de35b0/tests/SAML2/Response/SignatureValidationTest.php).

However we do not do this in our AuthnRequest factory, so I'm a bit on the fence on whether to keep it in or not.